### PR TITLE
MINOR: Fix broken version assertions in `ControllerChannelManagerTest`

### PR DESCRIPTION
**Pull request by hachikuji**
These assertions were broken for a number of reasons. Mainly the configuration wasn't being set correctly and we were overriding the intended version.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/controller/ControllerChannelManager.scala [hunk 1]
```diff
@@ -443,7 +443,8 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
 
   private def sendLeaderAndIsrRequest(controllerEpoch: Int, stateChangeLog: StateChangeLogger): Unit = {
     val leaderAndIsrRequestVersion: Short =
-      if (config.interBrokerProtocolVersion >= KAFKA_2_2_IV0) 2
+      if (config.interBrokerProtocolVersion >= KAFKA_2_3_IV1) 3
```
Comment (stanislavkozlovski) [inline core/src/main/scala/kafka/controller/ControllerChannelManager.scala:446]: The new LeaderAndIsr version was introduced with KIP-455, which would come out in 2.4. I believe we need to check against that version
https://github.com/apache/kafka/pull/7351#issuecomment-532267852

File: core/src/test/scala/unit/kafka/controller/ControllerChannelManagerTest.scala [hunk 2]
```diff
@@ -680,32 +689,29 @@ class ControllerChannelManagerTest {
       sentRequests(brokerId).append(SentRequest(request, callback))
     }
 
-    def collectStopReplicRequestsFor(brokerId: Int,
-                                     version: Short = ApiKeys.STOP_REPLICA.latestVersion): List[StopReplicaRequest] = {
+    def collectStopReplicRequestsFor(brokerId: Int): List[StopReplicaRequest] = {
```
Comment (stanislavkozlovski) [inline core/src/test/scala/unit/kafka/controller/ControllerChannelManagerTest.scala:692]: nit: typo `StopReplic`

File: core/src/test/scala/unit/kafka/controller/ControllerChannelManagerTest.scala [hunk 3]
```diff
@@ -605,8 +610,11 @@ class ControllerChannelManagerTest {
     assertEquals(1, batch.sentRequests.size)
     assertTrue(batch.sentRequests.contains(2))
 
-    val requests = batch.collectStopReplicRequestsFor(2, expectedStopReplicaRequestVersion)
-    assertTrue(requests.forall(_.version() == expectedStopReplicaRequestVersion))
+    val requests = batch.collectStopReplicRequestsFor(2)
```
Comment (stanislavkozlovski) [inline core/src/test/scala/unit/kafka/controller/ControllerChannelManagerTest.scala]: Given that we've verified right above that the `sentRequests` are just one, should we change this assertion to asserting just one request?

**Comment by hachikuji** [comment]
This patch will be incorporated in #7351. I will close this.

**Pull request closed without merging**